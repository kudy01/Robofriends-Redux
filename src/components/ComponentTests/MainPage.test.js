import React from "react";
import { shallow } from "enzyme";
import MainPage from "../MainPage";


let wrapper ;
beforeEach(() => {
	const mockProps = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: false
	}
	wrapper = shallow(<MainPage {...mockProps}/>)
})
it("Expect to render MainPage Component", () => {
  expect(wrapper).toMatchSnapshot(); 
}); 

it("Filters Robots Correctly when search found", () => {
	const mockProps2 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id: 3,
			name: 'John',
			email: 'jj'
		}],
		searchField: 'john',
		isPending: false
	}
	const wrapper2 = shallow(<MainPage {...mockProps2}/>)
	expect(wrapper2.instance().filterRobots()).toEqual([{
			id: 3,
			name: 'John',
			email: 'jj'
		}])
})

it("Filters Robots Correctly when search not found", () => {
	const mockProps3 = {
		onRequestRobots: jest.fn(),
		robots: [{
			id: 3,
			name: 'John',
			email: 'jj'
		}],
		searchField: 'jojo',
		isPending: false
	}
	const wrapper3 = shallow(<MainPage {...mockProps3}/>)
	expect(wrapper3.instance().filterRobots()).toEqual([])
})


it("Show the loading page", () => {
	const mockProps4 = {
		onRequestRobots: jest.fn(),
		robots: [],
		searchField: '',
		isPending: true
	}
	const wrapper4 = shallow(<MainPage {...mockProps4}/>)
	expect(wrapper4).toEqual({})
})