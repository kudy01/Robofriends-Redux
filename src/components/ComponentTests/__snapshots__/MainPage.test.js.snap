// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render MainPage Component 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
